presto-main/src/main/java/com/facebook/presto/operator/repartition/OptimizedPartitionedOutputOperator.java

```diff
@@ -494,7 +523,7 @@ public long getRetainedSizeInBytes()
             // When called in addInput(), the arrays have been returned to the arrayAllocator already,
             // but they're still owned by the decodedBlock which will be counted as part of the decodedBlock.
             // In both cases, the arrayAllocator doesn't need to be counted.
```

Comment:
this comment is now confusing (because bufferAllocator is an array allocator)

Reply:
@mbasmanova For now I'm using blockDecodingAllocator and bufferAllocator in PagePartitioner, and changed this comment to the following:
```
// When called by the operator constructor, the blockDecodingAllocator was empty at the moment.
            // When called in addInput(), the arrays have been returned to the blockDecodingAllocator already,
            // but they're still owned by the decodedBlock which will be counted as part of the decodedBlock.
            // In both cases, the blockDecodingAllocator doesn't need to be counted. But we need to count
            // bufferAllocator which contains buffers used during partitioning, e.g. serializedRowSizes,
            // mappedPositions, etc.
            long size = bufferAllocator.getEstimatedSizeInBytes();
```
Does this looks better? Or if you have better idea to count the memory using a single ArrayAllocator, we can remove this comment.